protocol/messenger_handler.go

```diff
@@ -499,6 +499,30 @@ func (m *Messenger) HandleSyncInstallationContact(state *ReceivedMessageState, m
 
 	}
 
+	// Sync last updated field
+	// We don't set `LastUpdated`, since that would cause some issues
+	// as `LastUpdated` tracks both display name & picture.
+	// The case where it would break is as follow:
+	// 1) User A pairs A1 with device A2
+	// 2) User B publishes display name and picture with LastUpdated = 3
+	// 3) Device A1 receives display name & picture with LastUpdated = 3
+	// 4) Device A1 syncs with A2 (which has not received the previous message)
+	// 5) Device A2 saves Display name and sets LastUpdated = 3,
+	//    note that picture has not been set as it's not synced
+	// 6) Device A2 receives the message from 2. Because LastUpdated is 3
```

Comment:
I read half a dozen times to really get the problem you're describing. Thank you, very useful comment :100: 

- Tiny typo on `6)`, "Because LastUpdated is 3" is duplicated.
- On step 3, it would've been easier to me if it was written: `3) Device A1 receives message from 2.`
- On step 4, it would've been a bit easier to follow with `4) Device A1 syncs with A2 (which has not received message from 3).`
- Nitpick: add the missing periods to sentences.